pediatric_rootlets/pediatric_rootlets.sh

```diff
@@ -0,0 +1,182 @@
+  # Update global variable with segmentation file name
+  FILELABEL="${file_t2}_labels-disc"
+  FILELABELMANUAL="${PATH_DATA}/derivatives/labels/${SUBJECT}/anat/${FILELABEL}.nii.gz"
+  echo "Looking for manual label: $FILELABELMANUAL"
+  if [[ -e $FILELABELMANUAL ]]; then
+    echo "Found! Using manual intervertebral disc labels."
+    rsync -avzh $FILELABELMANUAL ${FILELABEL}.nii.gz
+  else
+    echo "Manual intervertebral discs not found. Proceeding with automatic labeling."
+    # Generate labeled segmentation and disc labels
+    sct_label_vertebrae -i ${file_t2}.nii.gz -s ${FILESEG}.nii.gz -c t2 -qc ${PATH_QC} -qc-subject ${SUBJECT}
+    # Rename automatically generated disc labels to match the manual ones (otherwise, we would need conditionsto
+    # handle different filenames)
+    mv ${FILESEG}_labeled_discs.nii.gz ${FILELABEL}.nii.gz
+  fi
+  # Generate QC report for intervertebral disc labeling (either manual or automatic)
+  sct_qc -i ${file_t2}.nii.gz -s ${FILELABEL}.nii.gz -p sct_label_utils -qc ${PATH_QC} -qc-subject ${SUBJECT}
+}
+
+# SCRIPT STARTS HERE
+# ==============================================================================
+# Display useful info for the log, such as SCT version, RAM and CPU cores available
+sct_check_dependencies -short
+
+# Go to folder where data will be copied and processed
+cd $PATH_DATA_PROCESSED
+
+# Copy source images
+# Note: we copy only T2w to save space
+rsync -Ravzh ${PATH_DATA}/./${SUBJECT}/anat/${SUBJECT//[\/]/_}_*T2w.* .
+
+# Go to anat folder where all structural data are located
+cd ${SUBJECT}/anat
+
+# Define the names of the T2w files
+file_t2_composed=${SUBJECT}_rec-composed_T2w
+file_t2_top=${SUBJECT}_acq-top_run-1_T2w
+
+# Check if file_t2_composed exists, if not, use file_t2_top as file_t2
+if [ -f ${file_t2_composed}.nii.gz ]; then
+    file_t2=${file_t2_composed}
+
+# Use `acq-top` if the composed image does not exist. This is the case, for example, for `sub-125`.
+else
+    file_t2=${file_t2_top}
+    echo "Composed T2w file not found. Proceeding only with Top T2w file."
+fi
+
+ # Segment spinal cord (only if it does not exist)
+segment_if_does_not_exist ${file_t2}.nii.gz
+
+# Run sct_label_vertebrae for vertebral levels estimation
+label_if_does_not_exist ${file_t2}.nii.gz
+
+# Project the intervertebral disc labels to the spinal cord centerline
+sct_label_utils -i ${FILESEG}.nii.gz -o ${FILELABEL}_centerline.nii.gz -project-centerline ${FILELABEL}.nii.gz -qc ${PATH_QC} -qc-subject ${SUBJECT}
+
+# Detect PMJ (only if it does not exist)
+detect_pmj_if_does_not_exist ${file_t2}.nii.gz
+
+# Crop composed images and labels to the size of the top image (if the composed image exists)
+if [ -f ${file_t2_composed}.nii.gz ]; then
+  $SCT_DIR/python/envs/venv_sct/bin/python ~/code/model-spinal-rootlets/pediatric_rootlets/crop_composed_images.py -i_composed ${file_t2_composed}.nii.gz -s ${FILESEG}.nii.gz -d ${FILELABEL}_centerline.nii.gz -pmj ${FILEPMJ}.nii.gz -x 11
```

Comment:
Well, it turns out that running rootlets prediction on cropped images is not a good idea. Rootlets obtained from a cropped image have different S-I dimensions (fewer slices) than the original image. So I had to pad the images using relatively complex asymmetric padding (`sct_image -pad-asym`); see the script [here](https://data.neuro.polymtl.ca/datasets/philadelphia-pediatric/src/commit/677fe09f61e5974ccc1a2e8d69b9f1f084d3cdd4/code/pad_rootlets.sh). I then pushed the rootlets to our git-annex server in [this commit](https://data.neuro.polymtl.ca/datasets/philadelphia-pediatric/commit/677fe09f61e5974ccc1a2e8d69b9f1f084d3cdd4).
Based on these findings, I would eventually remove the cropping and continue working with the original images.